sky/data/skystorage.py

```diff
@@ -0,0 +1,227 @@
+
+
+@main.command()
+@click.argument('src', required=True, type=str)
+@click.argument('storetype', required=True, type=str)
+@click.argument('bucketname', required=True, type=str)
+@click.option('--num-threads', required=False, default=10, type=int, help='')
+@click.option('--interval', required=False, default=600, type=int, help='')
+@click.option('--delete',
+              required=False,
+              default=False,
+              type=bool,
+              is_flag=True,
+              help='')
+@click.option('--lock',
+              required=False,
+              default=False,
+              type=bool,
+              is_flag=True,
+              help='')
+@click.option('--no-follow-symlinks',
+              required=False,
+              default=False,
+              type=bool,
+              is_flag=True,
+              help='')
+def csync(src: str, storetype: str, bucketname: str, num_threads: int,
+          interval: int, delete: bool, lock: bool, no_follow_symlinks: bool):
+    """
+    Syncs the source to the bucket every INTERVAL seconds. Creates a lock file
+    while sync command is runninng and removes it when completed.
+    """
+    base_dir = os.path.expanduser(C_SYNC_FILE_PATH)
+    os.makedirs(base_dir, exist_ok=True)
+    lock_file_name = f'csync_{storetype}_{bucketname}.lock'
+    lock_path = os.path.expanduser(os.path.join(base_dir, lock_file_name))
+
+    # When the csync daemon was previously terminated abnormally,
+    # it may still be holding the lock. This lock will be reset after removal.
+    if os.path.exists(lock_path):
+        os.remove(lock_path)
+
+    while True:
+        with contextlib.ExitStack() as stack:
+            if lock:
+                stack.enter_context(filelock.FileLock(lock_path))
+            start_time = time.time()
+            # TODO: add try-except block
+            run_sync(src, storetype, bucketname, num_threads, interval, delete,
+                     no_follow_symlinks)
+            end_time = time.time()
+        # the time took to sync gets reflected to the INTERVAL
+        elapsed_time = int(end_time - start_time)
+        updated_interval = update_interval(interval, elapsed_time)
+        if lock and os.path.exists(lock_path):
+            os.remove(lock_path)
+        time.sleep(updated_interval)
+
+
+@main.command()
+def terminate() -> None:
+    """
+    Terminates all the CSYNC daemon running after checking if all the
+    sync process has completed.
+    """
+    # TODO: Currently, this terminates all the CSYNC daemon by default.
+    # Make an option of --all to terminate all and make the default
+    # behavior to take a source name to terminate only one daemon.
+    # Call the function to terminate the csync processes here
+    process_dict: Dict[int, Tuple[str, str]] = {}
+
+    # check all the running processes to see if CSYNC daemon is running
+    for proc in psutil.process_iter(['cmdline']):
+        cmd = proc.info['cmdline']
+        if 'csync' in cmd:
```

Comment:
This is dangerous, as it will cause any user commands with `csync` in it being killed by this. Is it possible that we can store the csync related information in a local db and load it from the disk when needed.

Reply:
Replaced `csync` with `skystorage csync`